examples/messaging/pub-sub/dotnet/Main.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using NATS.Client;
+
+// Create a new connection factory to create
+// a connection.
+Options opts = ConnectionFactory.GetDefaultOptions();
+opts.Url = "nats://nats:4222";
+
+// Creates a live connection to the default
+// NATS Server running locally
+ConnectionFactory cf = new ConnectionFactory();
+IConnection c = cf.CreateConnection(opts);
+
+// Setup an event handler to process incoming messages.
+// An anonymous delegate function is used for brevity.
+EventHandler<MsgHandlerEventArgs> h = (sender, args) =>
+{
+    // print the message
+    Console.WriteLine(args.Message);
+
+    // Here are some of the accessible properties from
+    // the message:
+    // args.Message.Data;
+    // args.Message.Reply;
+    // args.Message.Subject;
+    // args.Message.ArrivalSubcription.Subject;
+    // args.Message.ArrivalSubcription.QueuedMessageCount;
+    // args.Message.ArrivalSubcription.Queue;
+
+    // Unsubscribing from within the delegate function is supported.
+    args.Message.ArrivalSubcription.Unsubscribe();
+};
+
+// The simple way to create an asynchronous subscriber
+// is to simply pass the event in.  Messages will start
+// arriving immediately.
+IAsyncSubscription s = c.SubscribeAsync("foo", h);
+
+// Alternatively, create an asynchronous subscriber on subject foo,
+// assign a message handler, then start the subscriber.   When
+// multicasting delegates, this allows all message handlers
+// to be setup before messages start arriving.
+IAsyncSubscription sAsync = c.SubscribeAsync("foo");
+sAsync.MessageHandler += h;
+sAsync.Start();
+
```

Comment:
I'll remove this. While line 48 is executed, the subscription is made to the server, so the server starts sending messages. It's fast, but theoretically, you could miss messages.